From bc16688cd0ad17f2bbe8c224d39f1668c21e6e4b Mon Sep 17 00:00:00 2001
From: Wenzong Fan <wenzong.fan@windriver.com>
Date: Wed, 9 Mar 2016 02:58:55 -0500
Subject: [PATCH] wr-policy: fix avc denials for unix_chkpwd, systemd-*

* Allow chkpwd_t search /run/systemd/journal
* Allow systemd_tmpfiles_t read the network state information
* Allow systemd_passwd_agent_t read system state information in /proc

Fix avc denials:

  avc: denied { search } for pid=969 comm="unix_chkpwd" \
  name="journal" dev="tmpfs" ino=8203 \
  scontext=staff_u:staff_r:chkpwd_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:syslogd_var_run_t:s15:c0.c1023 \
  tclass=dir permissive=0

  avc: denied { read open getattr } for pid=798 comm="systemd-tmpfile" \
  name="unix" dev="proc" ino=4026532024 \
  scontext=system_u:system_r:systemd_tmpfiles_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:proc_net_t:s0 \
  tclass=file permissive=1

  avc: denied { read } for pid=971 comm="systemd-tty-ask" \
  name="filesystems" dev="proc" ino=4026532053 \
  scontext=system_u:system_r:systemd_passwd_agent_t:s0-s15:c0.c1023 \
  tcontext=system_u:object_r:proc_t:s0 \
  tclass=file permissive=0

Upstream-Status: Pending

Signed-off-by: Wenzong Fan <wenzong.fan@windriver.com>
---
 policy/modules/system/authlogin.te | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/policy/modules/system/authlogin.te b/policy/modules/system/authlogin.te
index 28f74ba..4b87358 100644
--- a/policy/modules/system/authlogin.te
+++ b/policy/modules/system/authlogin.te
@@ -99,7 +99,7 @@ allow chkpwd_t self:capability { dac_override setuid };
 dontaudit chkpwd_t self:capability sys_tty_config;
 allow chkpwd_t self:process { getattr signal };
 
-allow chkpwd_t shadow_t:file read_file_perms;
+allow chkpwd_t shadow_t:file { read_file_perms map };
 files_list_etc(chkpwd_t)
 
 kernel_read_crypto_sysctls(chkpwd_t)
@@ -113,6 +113,7 @@ dev_read_urand(chkpwd_t)
 dev_search_sysfs(chkpwd_t)
 
 files_read_etc_files(chkpwd_t)
+files_map_etc_files(chkpwd_t)
 # for nscd
 files_dontaudit_search_var(chkpwd_t)
 
@@ -138,6 +139,8 @@ seutil_dontaudit_use_newrole_fds(chkpwd_t)
 
 userdom_use_user_terminals(chkpwd_t)
 
+mls_file_read_all_levels(chkpwd_t)
+
 ifdef(`distro_ubuntu',`
 	optional_policy(`
 		unconfined_domain(chkpwd_t)
-- 
2.7.4

